### DISCUSSION TITLE
Development environment vector (build speed)

### DISCUSSION THREAD

**Discussion by dvob** [vectordotdev/vector#13392] [Q&A] [answered]
On my machine (32GB memory, i7-10510U CPU 1.80GHz) the build speed of vector is slow (1-2min).
I'm rather new to Rust so I rebuild my changes fairly often to understand if my changes would work. But if every build takes 1-2 minutes this is really painful.

Do others have the same problem? Is there a trick to speed up the build times? How does your development flow look like?

**Comment by jszwedko** [comment, marked as answer]
Hey @dvob !

Vector's compilation times are a bit of a known nuisance. We plan to make some improvements in this area soonish, but there are some tips in https://github.com/vectordotdev/vector/blob/master/docs/DEVELOPING.md#tips-and-tricks for doing Vector development like switching the linker and disabling features you aren't actively developing using. Hopefully this helps! I can definitely appreciate the slow build times being especially painful when you are new to Rust and relying on the compiler more to help guide development. I had the same experience coming to Vector.
